beacon-chain/core/light-client/lightclient.go

```diff
@@ -197,40 +267,251 @@ func NewLightClientFinalityUpdateFromBeaconState(
 				return nil, fmt.Errorf("invalid finalized header root %v", attestedState.FinalizedCheckpoint().Root)
 			}
 
-			finalizedHeader = &ethpbv2.LightClientHeader{Beacon: &ethpbv1.BeaconBlockHeader{
+			finalizedHeaderBeacon = &ethpbv1.BeaconBlockHeader{
 				Slot:          0,
 				ProposerIndex: 0,
 				ParentRoot:    make([]byte, 32),
 				StateRoot:     make([]byte, 32),
 				BodyRoot:      make([]byte, 32),
-			}}
+			}
 		}
 
 		var bErr error
 		finalityBranch, bErr = attestedState.FinalizedRootProof(ctx)
 		if bErr != nil {
-			return nil, fmt.Errorf("could not get finalized root proof %w", bErr)
+			return nil, errors.Wrap(bErr, "could not get finalized root proof")
 		}
 	} else {
-		finalizedHeader = &ethpbv2.LightClientHeader{Beacon: &ethpbv1.BeaconBlockHeader{
+		finalizedHeaderBeacon = &ethpbv1.BeaconBlockHeader{
 			Slot:          0,
 			ProposerIndex: 0,
 			ParentRoot:    make([]byte, 32),
 			StateRoot:     make([]byte, 32),
 			BodyRoot:      make([]byte, 32),
-		}}
+		}
 
 		finalityBranch = make([][]byte, FinalityBranchNumOfLeaves)
 		for i := 0; i < FinalityBranchNumOfLeaves; i++ {
 			finalityBranch[i] = make([]byte, 32)
 		}
 	}
 
-	result.FinalizedHeader = finalizedHeader
-	result.FinalityBranch = finalityBranch
+	switch block.Block().Version() {
+	case version.Altair, version.Bellatrix:
+		result.FinalizedHeader = &ethpbv2.LightClientHeaderContainer{
+			Header: &ethpbv2.LightClientHeaderContainer_HeaderAltair{
+				HeaderAltair: &ethpbv2.LightClientHeader{Beacon: finalizedHeaderBeacon},
+			},
+		}
+		result.FinalityBranch = finalityBranch
+	case version.Capella:
+		if finalizedBlock != nil && !finalizedBlock.IsNil() {
```

Comment:
`finalizedBlock != nil` is redundant here, `IsNil()` already does this

Reply:
if we don't check `finalizedBlock != nil` then the `!finalizedBlock.IsNil()` could throw a nil pointer dereference error